### PR TITLE
Dynamic syntax highlighting 

### PR DESCRIPTION
**Pull request by stmoy**
Integrating @oldnewthing's implementation to again support dynamic syntax highlighting. 

## Description

Get rid of the misleading RichTextBlock which are never shown to the user. Those are left over from earlier versions of the sample. Instead, there is a Xaml property whose value is a string that is the sample XAML. 

Dynamic content in the string is represented by $(...). Accompanying the string
is a collection of ControlExampleSubstitution objects, each of which has a Key and a Value. The key is the thing that goes between the parentheses, and the Value is the replacement.

The Value can be any object, and the code will try to pretty-print it in a XAML-friendly way:

* SolidColorBrush is represented by the underlying color.
* Null is represented by the empty string.
* Everything else becomes its own ToString().

Added a CSharp string property for sample C# code. Substitutions are performed in the Xaml, CSharp, and in the corresponding source files (XamlSource, CSharpSource).

Note that this change modifies the way that markup is represented in the sample and therefore will require a slightly new process for creating new samples. 

## Motivation and Context

#7 

## How Has This Been Tested?

Manual verification.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by oldnewthing** [comment]
> Get rid of the misleading which are never shown to the user.

I think a word missing.

File: XamlControlsGallery/ControlPages/ButtonPage.xaml [hunk 1]
```diff
@@ -31,9 +31,9 @@
                 </StackPanel>
             </local:ControlExample.Options>
             <local:ControlExample.Xaml>
-                <RichTextBlock Style="{StaticResource XamlCodeRichTextBlockStyle}">
-                    <Paragraph>&lt;Button Content="Button" Click="Button_Click" /&gt;</Paragraph>
-                </RichTextBlock>
+                <x:String>
+                    &lt;Button Content="Button" Click="Button_Click" /&gt;
```
Comment (YuliKl) [inline XamlControlsGallery/ControlPages/ButtonPage.xaml]: Should this sample code show the button's enabled/disabled state?

File: XamlControlsGallery/ControlPages/CalendarViewPage.xaml [hunk 2]
```diff
@@ -43,17 +43,22 @@
             </local:ControlExample.Options>
```
Comment (YuliKl) [inline XamlControlsGallery/ControlPages/CalendarViewPage.xaml]: This page still has the old language input (TextBox instead of ComboBox). I assume that change will be easy to merge from master when you check in this PR.

File: XamlControlsGallery/ControlPages/CommandBarPage.xaml [hunk 3]
```diff
@@ -42,16 +42,16 @@
...
-                    <Paragraph TextIndent="12">&lt;AppBarButton Icon="Add" Label="Add"/&gt;</Paragraph>
-                    <Paragraph TextIndent="12">&lt;AppBarButton Icon="ReShare" Label="Share"/&gt;</Paragraph>
-                    <Paragraph TextIndent="12">&lt;AppBarButton Icon="Edit" Label="Edit"/&gt;</Paragraph>
-                    <Paragraph TextIndent="12">&lt;CommandBar.SecondaryCommands&gt;</Paragraph>
-                    <Paragraph TextIndent="24">&lt;AppBarButton Icon="Setting" Label="Settings"/&gt;</Paragraph>
-                    <Paragraph TextIndent="12">&lt;/CommandBar.SecondaryCommands&gt;</Paragraph>
-                    <Paragraph>&lt;/CommandBar&gt;</Paragraph>
-                </RichTextBlock>
+                <x:String xml:space="preserve">
+&lt;CommandBar Background="Transparent" IsOpen="False" DefaultLabelPosition="Right"&gt;
```
Comment (YuliKl) [inline XamlControlsGallery/ControlPages/CommandBarPage.xaml]: Probably another good opportunity to make the sample dynamically update, but that change shouldn't be part of this PR.

File: XamlControlsGallery/ControlPages/PersonPicturePage.xaml [hunk 4]
```diff
@@ -32,18 +26,15 @@
...
-                    </RichTextBlock>
-                    <RichTextBlock x:Name="rtb1" Style="{StaticResource XamlCodeRichTextBlockStyle}" Visibility="{x:Bind DisplayNameRadio.IsChecked, Mode=OneWay}">
-                        <Paragraph>&lt;PersonPicture DisplayName=&quot;Jane Doe&quot; /&gt;</Paragraph>
-                    </RichTextBlock>
-                    <RichTextBlock x:Name="rtb2" Style="{StaticResource XamlCodeRichTextBlockStyle}" Visibility="{x:Bind InitialsRadio.IsChecked, Mode=OneWay}">
-                        <Paragraph>&lt;PersonPicture Initials=&quot;SB&quot; /&gt;</Paragraph>
-                    </RichTextBlock>
-                </Grid>
+                <x:String>
+                    &lt;PersonPicture $(ProfilePicture)$(DisplayName)$(Initials) /&gt;
```
Comment (YuliKl) [inline XamlControlsGallery/ControlPages/PersonPicturePage.xaml]: Should the sample be updating based on which radio button is selected? The code makes me think it should, but the app doesn't change its text.

File: XamlControlsGallery/ControlPages/RepeatButtonPage.xaml [hunk 5]
```diff
@@ -30,9 +30,9 @@
             </local:ControlExample.Options>
 
             <local:ControlExample.Xaml>
-                <RichTextBlock Style="{StaticResource XamlCodeRichTextBlockStyle}">
-                    <Paragraph>&lt;RepeatButton Content="Click and hold" Click="RepeatButton_Click" /&gt;</Paragraph>
-                </RichTextBlock>
+                <x:String>
+                    &lt;RepeatButton Content="Click and hold" Click="RepeatButton_Click" /&gt;
```
Comment (YuliKl) [inline XamlControlsGallery/ControlPages/RepeatButtonPage.xaml]: I'll open a new issue to track the missing IsEnabled property from sample

File: XamlControlsGallery/ControlPages/RevealFocusPage.xaml [hunk 6]
```diff
@@ -92,10 +92,9 @@
                 </StackPanel>
```
Comment (YuliKl) [inline XamlControlsGallery/ControlPages/RevealFocusPage.xaml]: This page is missing UIA names that I'm sure I added. Probably just haven't been merged in from Master.

File: XamlControlsGallery/ControlPages/ToggleButtonPage.xaml [hunk 7]
```diff
@@ -31,9 +31,9 @@
                 </StackPanel>
             </local:ControlExample.Options>
             <local:ControlExample.Xaml>
-                <RichTextBlock Style="{StaticResource XamlCodeRichTextBlockStyle}">
-                    <Paragraph>&lt;ToggleButton Content="ToggleButton" Click="Button_Click"/&gt;</Paragraph>
-                </RichTextBlock>
+                <x:String>
+                    &lt;ToggleButton Content="ToggleButton" Click="Button_Click"/&gt;
```
Comment (YuliKl) [inline XamlControlsGallery/ControlPages/ToggleButtonPage.xaml]: one more for future issue

File: XamlControlsGallery/ControlPages/TreeViewPage.xaml [hunk 8]
```diff
@@ -35,7 +35,7 @@
         </DataTemplate>
 
         <controlpages:ExplorerItemTemplateSelector 
-            x:Key="ExpolrerItemTemplateSelector"
+            x:Key="ExplorerItemTemplateSelector"
```
Comment (YuliKl) [inline XamlControlsGallery/ControlPages/TreeViewPage.xaml]: Thank you for fixing, that had been bugging me :)

File: XamlControlsGallery/ControlPages/PersonPicturePage.xaml (hunk 4 above)
Comment (oldnewthing) [reply XamlControlsGallery/ControlPages/PersonPicturePage.xaml]: It should. The `$(ProfilePicture)`, `$(DisplayName)` and `$(Initials)` should be dynamically updating based on the radio button. Let me take a closer look.

Comment (oldnewthing) [reply XamlControlsGallery/ControlPages/PersonPicturePage.xaml]: The markup says

    <local:ControlExampleSubstitution Key="ProfilePicture"
        IsEnabled="{x:Bind ProfileImageRadio.IsChecked.Value}"
         Value="..." />

and the IsEnabled should be changing as the selection changes.

Comment (YuliKl) [reply XamlControlsGallery/ControlPages/PersonPicturePage.xaml]: Not sure why that is, but I still see:
image

Comment (oldnewthing) [reply XamlControlsGallery/ControlPages/PersonPicturePage.xaml]: Investigating.

File: XamlControlsGallery/ControlPages/RepeatButtonPage.xaml (hunk 5 above)
Comment (YuliKl) [reply XamlControlsGallery/ControlPages/RepeatButtonPage.xaml]: https://github.com/Microsoft/Xaml-Controls-Gallery/issues/40

File: XamlControlsGallery/ControlPages/PersonPicturePage.xaml (hunk 4 above)
Comment (oldnewthing) [reply XamlControlsGallery/ControlPages/PersonPicturePage.xaml]: Duh. Missing `Mode=OneWay`  in the three `IsEnabled="{x:Bind …}"` attributes. Should be `IsEnabled="{x:Bind …, Mode=OneWay}"`.

File: XamlControlsGallery/ControlPages/CalendarViewPage.xaml (hunk 2 above)
Comment (YuliKl) [reply XamlControlsGallery/ControlPages/CalendarViewPage.xaml]: I was looking at the wrong branch. ComboBox is working beautifully and language is getting updated in the Xaml sample.

File: XamlControlsGallery/ControlPages/CommandBarPage.xaml (hunk 3 above)
Comment (YuliKl) [reply XamlControlsGallery/ControlPages/CommandBarPage.xaml]: https://github.com/Microsoft/Xaml-Controls-Gallery/issues/41

File: XamlControlsGallery/ControlPages/RevealFocusPage.xaml (hunk 6 above)
Comment (YuliKl) [reply XamlControlsGallery/ControlPages/RevealFocusPage.xaml]: I was looking at the wrong branch. The names are where they should be.

File: XamlControlsGallery/ControlPages/PersonPicturePage.xaml (hunk 4 above)
Comment (stmoy) [reply XamlControlsGallery/ControlPages/PersonPicturePage.xaml]: Fixed in latest commit.
